packages/medusa/src/types/common.ts

```diff
@@ -27,6 +27,11 @@ export interface FindConfig<Entity> {
 
 export type PaginatedResponse = { limit: number; offset: number; count: number }
 
+export type RepositoryQuery<T> = FindConfig<T> & {
```

Comment:
I don't think a new type is required at the moment since it is only used in the price-list repository and that the `withDeleted` is not used. I propose the move up the where as part of the repository method parameter typing. wdyt?

Reply:
I lifted it from your pr (https://github.com/medusajs/medusa/pull/1344) as the return type from buildQuery. Common/types might not be the place for it, but as I think we should make the type available in order to describe the result from buildQuery.

Reply:
Yeah I will move it to the common as you suggested so you will be able to use it as well 👍